src/uwtools/config/support.py

```diff
@@ -88,3 +89,17 @@ def represent(dumper: yaml.Dumper, data: TaggedString) -> yaml.nodes.ScalarNode:
         documentation for details.
         """
         return dumper.represent_scalar(data.tag, data.value)
+
+
+def represent_namelist(dumper: yaml.Dumper, data) -> yaml.nodes.MappingNode:
+    """
+    Serialize f90nml tags.
+    """
+    # Convert the f90nml Namelist object to a dictionary
+    namelist_dict = data.todict()
+    for key, value in namelist_dict.items():
+        if isinstance(value, OrderedDict):
+            namelist_dict[key] = dict(value)
```

Comment:
I was (perhaps naively) expecting this mod to have been in the `nml` module. Perhaps even the logic in the `__repr__` method of a nml config object, and the dumper function in the module. I'd be interested in hearing other thoughts on that.

Reply:
Hmm, my feeling was to try to keep `yaml` behavior in `yaml` methods, but I can see where you were going with that. `__repr__` might be the best place if we were to do that, I'll have to think about that one

Reply:
The current PR logic seems to me like a solution to the problem "How do I deal with unrepresentable `OrderedDict` objects when I dump a config to YAML?" It is independent of the fact that this situation is, at the moment, arising for us only in the context of `f90nml` `Namelist` objects -- it seems like it would handle/replace `OrderedDict` objects from any source at YAML dump time. That seems fine to me, if possibly more than what's strictly needed at the moment.

I think that maybe this routine belongs in `uwtools.config.formats.yaml` instead of in this module, which strikes me as too general -- would some code other than `uwtoolsconfig.formats.yaml` need this?

Also, does this correctly operate with any amount of nesting? Old code for dealing with `OrderedDict` objects in `f90nml` objects (see [here](https://github.com/ufs-community/uwtools/blob/965d8ddd3347777f5a5f7af8e8e22839ca79b8e5/src/uwtools/config/formats/nml.py#L36-L42)) operated recursively -- do we need to care about that here?

I don't immediately see why this would be related to `NMLConfig`'s `__repr__()` method: That would be a way to get a string representation of the object, but that's not necessarily YAML. In fact, shouldn't it be, well, the string representation of a namelist file, like `&nl n=88 /`, etc.? I could be missing something...

Reply:
Whats interesting is that afaik, `ini`'s implementation should create `OrderedDict`s, but `yaml` doesn't seem to mind. Testing with the other example `ini` files, something in the parser seems to step correctly instead of creating nested `OrderedDict`s. My wariness of putting it too high up was whether there was a condition that the native parser would handle correctly on its own but would generate errors if forced through. 

I don't think this comes up in any circumstance except when feeding non-`YAML` data directly to `yaml.dump()` so we could pre-parse it by default there?

Reply:
It sounds like they once used `OrderedDict` in `configparser` but eliminated it since all currently-support Python versions have an ordered `dict` by default: https://bugs.python.org/issue33504. Are you still seeing `OrderedDict` objects in values returned by `configparser`?

Reply:
So then the trick then would be to use configparser instead of the functions within f90nml. I had tried to do other handling within the yaml formatting, but ended up walking back to the dumper. The benefit should be in correctly parsing without having to use a conditional. 
The f90 Namelist object is a subclass of OrderedDict, so its handling was what I had been working through

Reply:
> So then the trick then would be to use configparser instead of the functions within f90nml.

No, sorry, that's not what I was getting at. I was just pointing out that `f90nml` may be the only place where `OrderedDict` object appear for us. Once upon a time that might have been an issue for `configparser` representations of INI data, but not these days.

> The benefit should be in correctly parsing without having to use a conditional.

Yes. I think I agree with your approach. It lets `YAMLConfig` do the right thing with an `OrderedDict` object _no matter how it got introduced_, with no conditional logic for handling `NMLConfig` objects separately. My only concerns were 1. Is the code in the right module? 2. Does it handle nesting/recursion correctly? 3. Are there any types other than `OrderedDict` that we need to be concerned with immediately? My best guess answers are: No, Yes, No.

I was chatting about this with @christinaholtNOAA in Slack, too, but since it's late in the day, maybe we can revisit this in DSM or Group tomorrow when we all have clear heads?

Reply:
AH, got it up on my big screen and I think I'm following properly now - I'll run a few tests through in the morning to ensure we have no spurious `OrderedDict` elsewhere (just to be sure), scoot this over to under `formats.yaml`, and fix that other namelist condition you suggested that's not working. Thanks a ton for the support!